### PR TITLE
Fix for issue #2

### LINKED ISSUE #2: Deuxieme prénom pas reconnu
Avec comme en-tête de fichier: 
/*
 \* Abc Def Ghi 1234567
*/
le script retourne: Étudiant trouvé: 'Abc Ghi' avec le matricule 1234567.

Expected: Étudiant trouvé: 'Abc Def Ghi' avec le matricule 1234567.

Je metterais un trait-d'union dans mon prénom mais il n'y en a pas dans mon dossier étudiant donc ça risque de poser problème.

### PR DESCRIPTION
**Pull request by SamuelYvon**
The tuples now contains the list of matched group. It works here because we know it is spaced separated but the python re lib really need to implements repeated group matches / by default repeated groups contain only the last instance. 

This is closing issue #2

I fixed the test cases for this issue. I was lazy and didn't do it properly first.

### CODE REVIEW

**Comment by SamuelYvon** [comment]
Also added a check to avoid crashing on starting the script without an internet connection

File: tests/test_correct.py [hunk 1]
```diff
@@ -66,12 +66,18 @@ def test_long_names(self):
 
         # the re library doesn't allow repeated group
         self.assertEqual(name_one[0], "author")
-        self.assertEqual(name_one[1], "tres")
-        self.assertEqual(name_one[2], "11112222")
+        self.assertEqual(name_one[1], "one")
```
Comment (Velythyl) [inline tests/test_correct.py]: Very good, shows support for bilinguals, from english to spanish

File: validate.py [hunk 2]
```diff
@@ -53,14 +58,12 @@ def extract_students(file: str):
...
 
-        matches = re.findall(
-            name_and_id_regex,
-            code)
-
         # Trim spaces
         names = []
-        for match in matches:
-            names.append(tuple((str(part).lstrip().rstrip() for part in match)))
+        for match in re.finditer(name_and_id_regex, code):
```
Comment (Velythyl) [inline validate.py]: Very good usage of re.finditer

**Review by Velythyl — APPROVED** [review]
LGTM! super good and follows current best practices in python - mixing ifs and exception checks is apparently the pythonic way to use Python.re /╲/\╭( ͡° ͡° ͜ʖ ͡° ͡°)╮/\╱\ (not even sarcastic here)

**Comment by SamuelYvon** [comment]
I can feel the python love I that review

**Pull request merged**